### DISCUSSION TITLE
Make attaching a Node more C# friendly

### DISCUSSION THREAD

**Discussion by kyle-chengzx** [godotengine/godot-proposals#8104] [Editor]
Hi, Recently started learning Godot using C#. I find when trying to attach the node the default name for the receiver method given is _on_item_function. Wondering if its possible to add like a C# toggle option to make it default in populating the name OnItemFunction instead?

**Comment by FarzadBT** [comment]
I'm gonna bump this with an example.
Godot's documentation for C# states that we should "[use PascalCase for all namespaces, type names and member level identifiers (i.e. methods...)](https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_style_guide.html#naming-conventions)" so it's a bit strange that the default receiver method proposed when you want to connect a signal to a C# script is written in snake_case, which is used for GDScript.
![image](https://github.com/godotengine/godot-proposals/assets/7903778/d4521c04-ebd2-4b2d-b702-520aea745b2e)
In this case, the expected receiver method would be `OnScoreTimerTimeout`
Is it possible to autoformat by using the script language of the node you're trying to connect to?
